07-routing-gossip.md

```diff
@@ -379,144 +371,231 @@ Note that such a `channel_update` that is not preceded by a `channel_announcemen
+  completed:
+    - SHOULD resume normal operation, as specified in the following
+    [Rebroadcasting](#rebroadcasting) section.
 
 ## Rebroadcasting
 
-Nodes receiving a new `channel_announcement` or a `channel_update` or
-`node_announcement` with an updated timestamp SHOULD update their local view of the network's topology accordingly.
-
-If, after applying the changes from the announcement, there are no channels associated with the announcing node, then the receiving node MAY purge the announcing node from the set of known nodes.
-Otherwise the receiving node updates the metadata and stores the signature associated with the announcement.
-This will later allow the receiving node to rebuild the announcement for its peers.
-
-Once the announcement has been processed, it is added to a list of outgoing announcements for the processing node's peers, perhaps replacing older updates. This list will be flushed at regular intervals.
-This store-and-delayed-forward broadcast is called a _staggered broadcast_
-
 ### Requirements
 
-Each node SHOULD flush outgoing announcements once every 60 seconds, independently of the arrival times of announcements, resulting in a staggered announcement and deduplication of announcements.
-
-Nodes MAY re-announce their channels regularly, however this is discouraged in order to keep the resource requirements low.
-
-Nodes SHOULD send all `channel_announcement` messages followed by the
-latest `node_announcement` and `channel_update` messages upon
-connection establishment.
+The final node:
+  - upon receiving a new `channel_announcement` or a `channel_update` or
+  `node_announcement` with an updated `timestamp`:
+    - SHOULD update its local view of the network's topology accordingly.
+  - after applying the changes from the announcement:
+    - if there are no channels associated with the corresponding origin node:
+      - MAY purge the origin node from its set of known nodes.
+    - otherwise:
+      - SHOULD update the appropriate metadata AND store the signature
+      associated with the announcement.
+        - Note: this will later allow the final node to rebuild the announcement
+        for its peers.
+
+An endpoint node:
+  - SHOULD flush outgoing announcements once every 60 seconds, independently of
+  the arrival times of announcements.
+    - Note: this results in staggered announcements that are unique (not
+    duplicated).
+  - MAY re-announce its channels regularly.
+    - Note: this is discouraged, in order to keep the resource requirements low.
+  - upon connection establishment:
+    - SHOULD send all `channel_announcement` messages, followed by the latest
+    `node_announcement` AND `channel_update` messages.
 
 ### Rationale
 
-Batching announcements forms a natural rate limit with low overhead.
+Once the announcement has been processed, it's added to a list of outgoing
+announcements, destined for the processing node's peers, and replaces any older
```

Comment:
Would "replacing any" be clearer? Right now you could read that as a separate action and I think it's the same.

Reply:
done